### PR TITLE
systemd-sysusers fsync patch

### PR DESCRIPTION
**Pull request by avylove**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
When fsync was added to systemd-sysusers in https://github.com/systemd/systemd/pull/6636, passwd file changes were missed. This can cause the passwd file to be left blank if a system has a hard shutdown early in the boot process. 

###### Change Log  

- fsync after updating passwd file with systemd-sysusers (https://github.com/systemd/systemd/pull/24324)

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- https://github.com/Azure/azure-sphere-os-scenario-tests/issues/40

###### Links to CVEs  
- N/A

###### Test Methodology

- Sphere pipeline: https://dev.azure.com/msazuresphere/Public/_build/results?buildId=277856&view=results

### CODE REVIEW

File: SPECS/systemd/systemd-bootstrap.spec [hunk 1]
```diff
@@ -12,6 +12,7 @@ Source1:        50-security-hardening.conf
 Source2:        systemd.cfg
 Source3:        99-dhcp-en.network
 Patch0:         fix-journald-audit-logging.patch
+Patch1:         add-fsync-sysusers-passwd.patch
```
Comment (PawelWMS) [inline SPECS/systemd/systemd-bootstrap.spec:15]: Thanks for the patch and porting it to Mariner! Let's also add a short comment here, so it's easier to track when this patch can be removed:

```suggestion
# Can be removed once we update systemd to a version containing the following commit:
# https://github.com/systemd/systemd/commit/19193b489841a7bcccda7122ac0849cf6efe59fd
Patch1:         add-fsync-sysusers-passwd.patch
```

File: SPECS/systemd/systemd.spec [hunk 2]
```diff
@@ -12,6 +12,7 @@ Source1:        50-security-hardening.conf
 Source2:        systemd.cfg
 Source3:        99-dhcp-en.network
 Patch0:         fix-journald-audit-logging.patch
+Patch1:         add-fsync-sysusers-passwd.patch
```
Comment (PawelWMS) [inline SPECS/systemd/systemd.spec:15]: Same as in the other spec:
```suggestion
# Can be removed once we update systemd to a version containing the following commit:
# https://github.com/systemd/systemd/commit/19193b489841a7bcccda7122ac0849cf6efe59fd
Patch1:         add-fsync-sysusers-passwd.patch
```

**Review by PawelWMS — APPROVED** [review]
Appreciate the fix @avylove! Approving but please incorporate the two comments I've mentioned. Thanks!

**Comment by avylove** [comment]
Added removal comments to spec files

**Pull request merged**